meilisearch-core/src/error.rs

```diff
@@ -33,6 +37,26 @@ impl From<io::Error> for Error {
     }
 }
 
+impl From<PestError<Rule>> for Error {
+    fn from(error: PestError<Rule>) -> Error {
+        Error::FilterParseError(error.renamed_rules(|r| {
+            let s  = match r {
+                Rule::or => "OR",
+                Rule::and => "AND",
+                Rule::not => "NOT",
+                Rule::string => "string",
+                Rule::word => "word",
+                Rule::greater => "field>value",
+                Rule::lesser => "field<value",
```

Comment:
Don't you forget to add `geq` and `leq`?

Reply:
I would prefer you list all the variants in the last `_` other branch, this way we avoid missing any.

Reply:
yes indeed, I forgot them.

Many rules are actually silent, and never generate tokens, for all rules but those two, there is a rewrite rule.